File: 백준/Silver/1406. 에디터/에디터.py
from collections import deque

arr = list(input())
stack = deque()

N = int(input())
for i in range(N):
    ar = input().split()
    if ar[0] == 'L' and arr:
        stack.appendleft((arr.pop()))
    if ar[0] == 'D' and stack:
        arr.append(stack.popleft())
    if ar[0] == 'B' and arr:
        arr.pop()
    if ar[0] == 'P':
        arr.append(ar[1])


for i in arr:
    print(i,end='')
for i in stack:
    print(i,end='')

File: 백준/Silver/1406. 에디터/에디터.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;


public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String str = br.readLine();
        Stack<Character> left = new Stack<>();
        Stack<Character> right = new Stack<>();


        for (int i=0; i<str.length(); i++){
            left.push(str.charAt(i));
        }

        String command;

        int n = Integer.parseInt(br.readLine());
        for (int i=0; i<n; i++) {
            String[] commandArr = br.readLine().split(" ");
            command = commandArr[0];
            switch (command) {
                case "L":
                    if (left.empty()) continue;
                    right.push(left.pop());
                    break;
                case "D":
                    if (right.empty()) continue;
                    left.push(right.pop());
                    break;
                case "B":
                    if (left.empty()) continue;
                    left.pop();
                    break;
                case "P":
                    left.push(commandArr[1].charAt(0));
                    break;
            }
        }

        StringBuilder sb = new StringBuilder();
        while (!left.isEmpty()) {
            right.push(left.pop());
        }
        while (!right.isEmpty()) {
            sb.append(right.pop());
        }
        System.out.print(sb);
        br.close();
    }
}
